src/Ledger/Certs/Properties.agda

```diff
@@ -131,3 +133,115 @@ instance
 
 Computational-CERTS : Computational _⊢_⇀⦇_,CERTS⦈_ String
 Computational-CERTS = it
+
+private variable
+  dCert : DCert
+  Γ : CertEnv
+  l : List DCert
+
+instance
+  _ = +-0-monoid
+
+  HasCoin-Map : ∀ {A} → ⦃ DecEq A ⦄ → HasCoin (A ⇀ Coin)
+  HasCoin-Map .getCoin s = ∑[ x ← s ] x
+
+≡ᵉ-getCoin : ∀ {A} → ⦃ _ : DecEq A ⦄ → (s s' : A ⇀ Coin) → s ˢ ≡ᵉ s' ˢ → getCoin s ≡ getCoin s'
+≡ᵉ-getCoin {A} ⦃ decEqA ⦄ s s' s≡s' = indexedSumᵛ'-cong {C = Coin} {x = s} {y = s'} s≡s'
+
+module _ {A : Type} ⦃ _ : DecEq A ⦄
+  where
+  getCoin-singleton : ((a , c) : A × Coin) → indexedSumᵛ' id ❴ (a , c) ❵ ≡ c
+  getCoin-singleton _ = indexedSum-singleton' ⦃ M = +-0-commutativeMonoid ⦄ (finiteness _)
+
+  module _ {gc-hom : (d₁ d₂ : A ⇀ Coin) → getCoin (d₁ ∪ˡ d₂) ≡ getCoin d₁ + getCoin d₂} where
+    ∪ˡsingleton≡ : {m : A ⇀ Coin} {(a , c) : A × Coin} → getCoin (m ∪ˡ ❴ (a , c) ❵ᵐ) ≡ getCoin m + c
```

Comment:
This is only true if `a ∉ dom m`, otherwise the rhs is going to be `getCoin (m ∣ ❴ a ❵ ᶜ) + c` (which it also is in the other case actually). This follows from the wrong assumption, see below.